lib/sprockets/media_query_combiner/processor.rb

```diff
@@ -1,15 +1,40 @@
-require 'tilt'
 require 'sass/media_query_combiner/combiner'
 
 module Sprockets
   module MediaQueryCombiner
-    class Processor < Tilt::Template
-      def prepare
+    
+    # serves to pick proper processor
+    class Processor
+      # returns processor class valid for current Sprockets version
+      def self.choose
+        return SprocketsProcessor if Sprockets.respond_to? :register_postprocessor
+        TiltProcessor
       end
+    end
 
-      def evaluate(context, locals, &block)
-        Sass::MediaQueryCombiner::Combiner.combine(data)
+    # for Sprockets >= 3.X
+    class SprocketsProcessor
+      def self.call(input)
+        Sass::MediaQueryCombiner::Combiner.combine input[:data]
       end
     end
+
+    # support for older Sprockets versions (<3.X)
+    begin
+      require 'tilt'
+
+      class TiltProcessor < Tilt::Template
+        def prepare
+        end
+
+        def evaluate(context, locals, &block)
+          Sass::MediaQueryCombiner::Combiner.combine data
+        end
+      end
+    
+    rescue LoadError
+      STDOUT.puts "Using native Sprockets processor"
```

Comment:
We should definitely be spamming stdout for things like this. Could you refactor this whole thing to conditionally define a single class based on sprockets version. There should be no `rescue` or `choose` method, you'd just use the class name. Something like:

```ruby
if # ... 
  class Processor
    # ...
  end
else
  require 'tilt'
  class Processor
    # ...
  end
end